feature-libs/order/occ/adapters/occ-order-history-extended.adapter.ts

```diff
@@ -0,0 +1,162 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { forkJoin, Observable, of } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+import {
+  Order,
+  Consignment,
+  OrderHistoryList,
+  ReturnRequestList,
+  OrderHistory,
+  ConsignmentTracking,
+} from '@spartacus/order/root';
+import { OccOrderHistoryAdapter } from './occ-order-history.adapter';
+
+@Injectable()
+export class OccOrderHistoryExtendedAdapter extends OccOrderHistoryAdapter {
+  /**
+   * fills the tracking information for each consignment in the input order details
+   * @param userId user id
+   * @param order order details with consignments
+   * @returns order details along with tracking information for each of it's consignments
+   */
+  fillConsignmentTrackingDetails(
+    userId: string,
+    order: Order
+  ): Observable<Order> {
+    const requests = order.consignments?.map((consignment: Consignment) => {
+      if (consignment.code && consignment.trackingID) {
+        return this.getConsignmentTracking(
+          order?.code ?? '',
```

Comment:
[MINOR]
I'm not fond of tricking type safety by fallbacking to a dummy value (e.g. `''`) just for the code to compile, but won't work correctly in runtime. Instead I suggest to first check: `if(!order?.code) { throw new Error(<short meaningful error message here)> }`

Reply:
the 'code' for any order is the key to identify that order uniquely. There is no way that can be empty functionally. hence I added ?? to just satisfy the checks, and did not throw any errors messages